### PR TITLE
refactor(accessible-button): to typescript

### PR DESCRIPTION
**Pull request by adnasa**
#### Summary

- migrates `accessible-button` to typescript.
- This is needed for the input-utils

### CODE REVIEW

**Comment by emmenko** [comment]
I'll take over this

File: packages/utils/src/invariant.ts [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+
 const isProduction: boolean = process.env.NODE_ENV === 'production';
 // @TODO: allow consumer to set the prefix e.g `[ Text.Body ]`
 const prefix: string = 'Invariant failed';
 
 // Throw an error if the condition fails
 // Strip out error messages for production
 // > Not providing an inline default argument for message as the result is smaller
-const invariant = (condition: unknown, message?: string): asserts condition => {
+const invariant: TInvariantFunction = (condition, message) => {
```
Comment (tdeekens) [inline packages/utils/src/invariant.ts:16]: Isn't this renamed to warning by now?

**Review by tdeekens — COMMENTED** [review]
😍

File: packages/utils/src/invariant.ts (hunk 1 above)
Comment (emmenko) [reply packages/utils/src/invariant.ts]: Ah yes, I've rebased the branch now.

**Comment by emmenko** [comment]
FYI: notice that the "README docs" are in a `docs_disabled` folder. The reason is that the current generator setup (react-docgen) throws an error because of the usage of `forwardRef`, which the `setParamsTypeDefinitionFromFunctionType` custom handler does not support apparently.

https://github.com/commercetools/ui-kit/blob/96c0718a4c5b06ef6f3b70b78ab16fc30186fefa/generators/readme/src/index.ts#L325

I think we should try looking into another library that has better docgen support for typescript, for example https://github.com/styleguidist/react-docgen-typescript

**Review by tdeekens — APPROVED** [review]

**Comment by adnasa** [comment]
> I think we should try looking into another library that has better docgen support for typescript, for example https://github.com/styleguidist/react-docgen-typescript


okay, good call and thanks for looking into it
I will merge this PR and follow up on the docgen.

**Comment by emmenko** [comment]
Alright, so the real error is this:

image

I'll push the fix.

**Comment by emmenko** [comment]
So this is tricky. For some reason the babel plugin transforms `React.ElementType` to `PropTypes.node`, instead of `PropTypes.elementType` which is what I would expect.

image

The prop type warning is caused by passing an element type like `as={Link}`, which is not a `node`.

**Pull request merged**